pkg/config/spec.go

```diff
@@ -562,45 +609,10 @@ func GetSourceSize(config *Config, source *v1.ImageSource) (int64, error) {
 		size, err = config.ImageExtractor.GetOCIImageSize(source.Value(), config.Platform.String())
 		size = int64(float64(size) * 2.5)
 	case source.IsDir():
-		err = fsutils.WalkDirFs(config.Fs, source.Value(), func(path string, d fs.DirEntry, err error) error {
-			if err != nil {
-				return err
-			}
-			if d.Type()&fs.ModeSymlink != 0 {
-				// If it's a symlink, get its target and calculate its size.
-				linkTarget, err := os.Readlink(path)
-				if err != nil {
-					return err
-				}
-
-				if !filepath.IsAbs(linkTarget) {
-					// If it's a relative path, join it with the base directory path.
-					linkTarget = filepath.Join(filepath.Dir(path), linkTarget)
-				}
-
-				_, err = os.Stat(linkTarget)
-				if os.IsNotExist(err) {
-					size += 0
-				} else if err != nil {
-					return err
-				} else {
-					linkInfo, err := os.Stat(linkTarget)
-					if err != nil {
-						return err
-					}
-					size += linkInfo.Size()
-				}
-			} else if !d.IsDir() {
-				// If it's a regular file, add its size to the total.
-				fileInfo, err := d.Info()
-				if err != nil {
-					return err
-				}
-
-				size += fileInfo.Size()
-			}
+		filesVisited = make(map[string]bool)
 
-			return nil
+		err = fsutils.WalkDirFs(config.Fs, source.Value(), func(path string, d fs.DirEntry, err error) error {
+			return getDirSize(size, filesVisited, path, d, err)
```

Comment:
How can this function change the value of `size`? It's not a pointer, it's just [an int64](https://github.com/kairos-io/kairos-agent/pull/168/files#diff-53b724197bc14c3828ce6725322b892cb788860b0526184b63d7d114edb60941R600). What am I missing?

Reply:
in this case size is an accumulator that gets passed. WalkDirFs will call the function every time for a new file updating the value